docs/ADR/013-transformation-services-infra.md

```diff
@@ -0,0 +1,69 @@
+# ADR-013-transformation-services-infra
+
+* Creation Date: 28/06/22
+* Status: Pending
+
+## Context
+
+As we start our deployment on Squarescale, we have to start thinking about the scalability and the dependencies of our services. For now, our API cannot work without our transformation services ready. We wish to separate them so they can launch without one another and recover if their counterpart crash. Also, we can add new transformation services without restarting the entire app.
+
+Adress are handled by Squarescale so we have to rely on Consul to fetch the list of existing services or have the services announce themselve at launch.
+
+## Decision
+
+
+## Options - Services registration
+
+### 1. Consul take the lead
+
+Consul is in charge of networking automation. As such, we may be able to add an event listener and get it to send an updated list of the transformation services when a new transformation service is started/stopped.
+![Diagram Option 1](../diagram/transformers_discovery/ADR_13_opt1.jpg)
+
+#### Benefits
+- Act only when needed.
+- Detection of a service absence/crash is left to Consul.
+- We don't need to reconnect to the service with each request.
+
+#### Drawbacks
+- Not sure it's Consul task.
+- Need a cache to keep the list in the API.
+- In case of multiple API, Consul may update only one per request.
+- Harder implementation.
```

Comment:
+ We need to extend consul for that. do we want to take part of Consul devlopment and buy all problems this wonderful projest already has ?